src/main/java/seedu/contax/logic/commands/RangeCommand.java

```diff
@@ -0,0 +1,114 @@
+            + "Parameters: from/FROM_INDEX to/TO_INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com"
+            + PREFIX_RANGE_FROM + "1"
+            + PREFIX_RANGE_TO + "3";
+
+    private final Index fromIndex;
+    private final Index toIndex;
+    private final ArgumentMultimap argMultimap;
+    private final String commandInput;
+    /**
+     * @param fromIndex              of the person in the filtered person list to edit
+     * @param toIndex                of the person in the filtered person list to edit
+     * @param argMultimap            details to argMultimap
+     * @param commandInput            details to word of command
+     */
+    public RangeCommand(Index fromIndex, Index toIndex, ArgumentMultimap argMultimap, String commandInput) {
+        requireNonNull(fromIndex);
+        requireNonNull(toIndex);
+        requireNonNull(argMultimap);
+        this.fromIndex = fromIndex;
+        this.toIndex = toIndex;
+        this.argMultimap = argMultimap;
+        this.commandInput = commandInput.trim();
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (fromIndex.getZeroBased() > toIndex.getZeroBased()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+        List<CommandResult> commandResultList = new ArrayList<>();
+        for (int i = fromIndex.getOneBased(); i <= toIndex.getOneBased(); i++) {
+            AddressBookParser addressBookParser = new AddressBookParser();
+            try {
+                String commandText = createNewCommand(commandInput, Integer.toString(i));
+                if (argMultimap.getPreamble().equals("delete")) {
+                    commandText = createNewCommand(commandInput, Integer.toString(fromIndex.getOneBased()));
+                }
+                Command command = addressBookParser.parseCommand(commandText);
+                commandResultList.add(command.execute(model));
+            } catch (ParseException pe) {
+                return new CommandResult(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+            }
+        }
+        StringBuilder resultOutput = new StringBuilder();
+        for (CommandResult result: commandResultList) {
+            resultOutput.append(result.getFeedbackToUser()).append("\n");
+        }
+        return new CommandResult(resultOutput.toString());
+    }
+
+    private String createNewCommand(String commandInput, String index) {
+        StringBuilder output = new StringBuilder();
+        String[] splitCommand = commandInput.trim().split(" ");
+        output.append(splitCommand[0]).append(" ").append(index).append(" ");
+        for (int i = 1; i < splitCommand.length; i++) {
+            output.append(" ").append(splitCommand[i]);
+        }
```

Comment:
This part could be simpler:
```suggestion
        output.append(commandInput.substring(splitCommand[0]);
```

Reply:
is substring para shd be index int instead of string

Reply:
anyway thx and change to oneline code